crates/router/src/core/payments.rs

```diff
@@ -4053,12 +4059,16 @@ where
     let (profile_id, routing_algorithm_id) = {
         let (profile_id, routing_algorithm) = match &transaction_data {
             TransactionData::Payment(payment_data) => (
-                payment_data.payment_intent.profile_id.clone(),
+                payment_data
```

Comment:
we have profile_id from business_profile which is taken in the arguments

Reply:
@kashif-m , can you confirm this for payout attempt's profile id as well. And if so was there any specific reason we decided to pass payment_data in Transaction type enum .

Reply:
It's the same for payouts as well, we can use `profile_id` in business profile

We are passing payment_data or payout_data in Transaction enum mainly for making dsl_input